packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/EnumProviderSerializationTests.cs

```diff
@@ -20,43 +20,71 @@ public void Setup()
             {
-                new InputEnumTypeValue(stringA, intA, null),
-                new InputEnumTypeValue(stringB, intB, null)
+                new InputEnumTypeValue("One", 1f, null),
+                new InputEnumTypeValue("Two", 2f, null)
             };
-            var input = new InputEnumType("mockInputEnum", "mockNamespace", "public", null, "The mock enum", InputModelTypeUsage.Input | InputModelTypeUsage.Output, InputPrimitiveType.Int32, values, false);
-            TypeProvider enumType = ClientModelPlugin.Instance.TypeFactory.CreateEnum(input);
-            return enumType.SerializationProviders.FirstOrDefault();
-        }
+            var floatType = new InputEnumType("mockInputEnum", "mockNamespace", "public", null, "The mock enum", InputModelTypeUsage.Input | InputModelTypeUsage.Output, InputPrimitiveType.Float32, floatValues, false);
 
-        [TestCaseSource(nameof(ValidateTestCases))]
-        public void ValidateToSerial(string stringA, int intA, string stringB, int intB)
-        {
-            var serialization = CreateEnumSerializationProvider(stringA, intA, stringB, intB);
-            MethodProvider? method = serialization!.Methods.Where(m => m.Signature.Name.Contains("ToSerial")).FirstOrDefault();
-            // Cast method.BodyExpression to SwitchCaseExpression
-            Assert.IsNull(method);
+            var stringValues = new List<InputEnumTypeValue>
+            {
+                new InputEnumTypeValue("One", "1", null),
+                new InputEnumTypeValue("Two", "2", null)
+            };
+            var stringType = new InputEnumType("mockInputEnum", "mockNamespace", "public", null, "The mock enum", InputModelTypeUsage.Input | InputModelTypeUsage.Output, InputPrimitiveType.String, stringValues, false);
+
+            return [intType, floatType, stringType];
         }
 
-        [TestCaseSource(nameof(ValidateTestCases))]
-        public void ValidateToEnum(string stringA, int intA, string stringB, int intB)
+        [TestCaseSource(nameof(ValidateTypes))]
+        public void ValidateToEnumMethods(InputEnumType inputEnum)
         {
-            var serialization = CreateEnumSerializationProvider(stringA, intA, stringB, intB);
+            TypeProvider enumType = ClientModelPlugin.Instance.TypeFactory.CreateEnum(inputEnum);
+            var serialization = enumType.SerializationProviders.FirstOrDefault();
             MethodProvider? method = serialization!.Methods.Where(m => m.Signature.Name.Contains("Enum")).FirstOrDefault();
             // Cast method.BodyExpression to SwitchCaseExpression
             if (method!.BodyStatements is MethodBodyStatements methodBodyStatements)
             {
-                // Verify that the switch case expression has the correct number of cases (values + 1 for throw)
+                // Verify that there are the correct number of cases (values + 1 for throw)
                 Assert.AreEqual(3, methodBodyStatements.Statements.Count());
```

Comment:
should not hard code 3, it should be inputEnum.Values.Count + 1 right?  If one of your items in your test case source has more than 2 values this will fail